### PR TITLE
add the test data and script

### PR DESCRIPTION
**Pull request by Zzzzzhuzhiwei**
In this issue #1266, I get the data of  these seven metrics, and write the script to make sure the data is right.

### CODE REVIEW

File: test/data.test.js [hunk 1]
```diff
@@ -0,0 +1,77 @@
+var openDigger = require('../src/open_digger')
+const fs = require('fs');
+const assert = require('assert');
+const _ = require('lodash');
+const _isEqual = require('lodash/isEqual');
+const FILENAME = '../test/testdata.json';
```
Comment (frank-zsy) [inline test/data.test.js:6]: For flexibility concern, I would like to suggest another file structure.

- Put all the data to `testdata` folder.
- Use `dataKey` as the file name of each data.
- In the file, we could provide a valid JSON object with the result and also the SQL to generate the result.

With above structure, we can add lots of test data in different files and easily to check if the data is right or not as well as the corresponding SQL.

Comment (Zzzzzhuzhiwei) [reply test/data.test.js:6]: I roughly understand, which folder is the `testdata` folder under?

**Comment by Zzzzzhuzhiwei** [comment]
Now I put the `testdata` folder under the `test` folder and first generate the SQL and results of `issueClosed`.

**Comment by frank-zsy** [comment]
Yes, this is exactly that I mean. And we definitely should design the naming rule here. I think we can all use underscore naming style. And all the parameters should be included in the file name to make sure no file data conflicts.

**Comment by Zzzzzhuzhiwei** [comment]
Do the file names and parameters in the file both need to use underscore naming style，and does the file name need to contain the parameters: OrgID, limit, startyear, startmonth, endyear, endmonth and groupTimeRange?

**Comment by frank-zsy** [comment]
> Do the file names and parameters in the file both need to use underscore naming style，and does the file name need to contain the parameters: OrgID, limit, startyear, startmonth, endyear, endmonth and groupTimeRange?

It depends on what we need to test. Org id is not necessary, year and month may not need to be included, just set to a constant is OK. But groupTimeRange is surely needed, and order, limit, limit type, group type(repo, org, label) are all needed.

**Comment by Zzzzzhuzhiwei** [comment]
`issueClosed_desc_3_all_repo_month.json` ,from front to back are metric, order, limit, limitoption, groupby, groupTimeRange. Is this file naming format acceptable？

**Comment by frank-zsy** [comment]
`issue_closed_...` is better, I think the metrics naming can be changed to underscore style too.

**Comment by Zzzzzhuzhiwei** [comment]
I will temporarily set the `order` to `desc` and the `limitOption` to `all` to generate the metric results, in order to avoid uncertainty in the results.

**Comment by frank-zsy** [comment]
That's OK, actually for some metrics with the problem, we can skip for now.

**Comment by Zzzzzhuzhiwei** [comment]
I have added the testdata with determined results into files.

**Comment by Zzzzzhuzhiwei** [comment]
Hi @frank-zsy , could you please review my PR? Your feedback would be greatly appreciated!

**Comment by frank-zsy** [comment]
@Zzzzzhuzhiwei Sure, I will review this PR today.

**Comment by frank-zsy** [comment]
@Zzzzzhuzhiwei Actually we are already now using `mocha` and `ts` extension to run the unit tests, so can you change the `js` file to `ts`, because right now your test file are not running in `npm test` command.

**Comment by Zzzzzhuzhiwei** [comment]
Thank you for your suggestion, I will change the `js` file to `ts` today.

**Comment by Zzzzzhuzhiwei** [comment]
hi @frank-zsy, I have changed the `js` file to `ts`, and my test file are running in `npm test` command. Thank you for your guidance very much!

**Comment by frank-zsy** [comment]
@Zzzzzhuzhiwei The result looks good, I think you can delete the old `metrics.test.ts` and rename the new file to `metrics.test.ts`.

File: test/metrics.test.ts [hunk 2]
```diff
@@ -1,210 +1,316 @@
...
-    // order works fine
-    if (options?.index !== undefined) {
-      assert.strictEqual(parseFloat(result[0][key][years - 1][options.index]) >= parseFloat(result[option.limit - 1][key][years - 1][options.index]), true);
-    } else {
-      assert.strictEqual(parseFloat(result[0][key][years - 1]) >= parseFloat(result[option.limit - 1][key][years - 1]), true);
+function deepEqual(a: any, b: any): boolean {
+  return _.isEqual(a, b);
+}
+
+async function validate_data(
```
Comment (frank-zsy) [inline test/metrics.test.ts:10]: We should keep identical naming style, the function name should be `validateData`.

**Comment by frank-zsy** [comment]
All looks good now, thanks for the contribution~

/approve

**Pull request merged**